File: Backend/src/models/boilerplate/faq.model.js
const mongoose = require('mongoose');
const { toJSON, paginate } = require('../plugins');

const faqSchema = mongoose.Schema(
  {
    question: {
      type: String,
      trim: true,
      default: null,
    },
    answer: {
      type: String,
      trim: true,
      default: null,
    },
    category: {
      type: String,
      trim: true,
      default: null,
    },
    status: {
      type: Boolean,
      default: true,
    },
    language: {
      type: mongoose.Schema.Types.ObjectId,
      default: null,
    },
    clientId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Client',
    },
    zoneId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Zone',
    },
    /** Who should see this FAQ: passenger app, driver app, or both (legacy docs without this field behave like Both). */
    userType: {
      type: String,
      enum: ['User', 'Driver', 'Both'],
      default: 'Both',
    },
  },
  {
    timestamps: {
      createdAt: 'createdAt',
      updatedAt: 'updatedAt',
    },
  },
);

// Add plugin that converts mongoose to json
faqSchema.plugin(toJSON);
faqSchema.plugin(paginate);

/**
 * @typedef Faq
 */
const Faq = mongoose.model('Faq', faqSchema);

module.exports = Faq;

File: Driver/lib/network/response_models/faq_model.dart
class FaqModel {
  String? question;
  String? answer;
  String? category;
  bool? status;
  String? language;
  String? clientId;
  String? id;

  FaqModel(
      {this.question,
        this.answer,
        this.category,
        this.status,
        this.language,
        this.clientId,
        this.id});

  FaqModel.fromJson(Map<String, dynamic> json) {
    question = json['question'];
    answer = json['answer'];
    category = json['category'];
    status = json['status'];
    language = json['language'];
    clientId = json['clientId'];
    id = json['id'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['question'] = question;
    data['answer'] = answer;
    data['category'] = category;
    data['status'] = status;
    data['language'] = language;
    data['clientId'] = clientId;
    data['id'] = id;
    return data;
  }
}